transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -350,14 +350,9 @@ private void closeAll() {
         } catch (IOException ignore) {
             // ignore on close
         }
-        Collection<AbstractEpollChannel> array = new ArrayList<AbstractEpollChannel>(channels.size());
-
-        for (AbstractEpollChannel channel: channels.values()) {
-            array.add(channel);
-        }
-
-        for (AbstractEpollChannel ch: array) {
-            ch.unsafe().close(ch.unsafe().voidPromise());
+        for (PrimitiveEntry<AbstractEpollChannel> entry : channels.entries()) {
```

Comment:
Perhaps this is a rudiment, preserved from the initial commit:

https://github.com/fenik17/netty/blob/9330172f803f340df677c370464126cd6112204a/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java#L292

also a `channels` map is not thread-safe.